### PR TITLE
[gameplaykit] Add nullability to (generated and manual) bindings

### PR DESCRIPTION
**Pull request by tj-devel709**
This PR aims to bring nullability changes to GameplayKit.
Following the steps [here](https://github.com/xamarin/xamarin-macios/wiki/Nullability):

1. I am adding `nullable enable` to all manual files that are not "API_SOURCES" in src/frameworks.sources and making the required nullability changes
2. Changing all `throw new ArgumentNullException ("object"));` to `ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (object));` for size saving optimization as well to mark that this framework contains nullability changes
3. Changing any `== null` or `!= null` to `is null` and `is not null`

### CODE REVIEW

File: src/GameplayKit/GKBehavior.cs [hunk 1]
```diff
@@ -18,7 +20,7 @@ public partial class GKBehavior {
 		}
 
 		public NSNumber this [GKGoal goal] {
-			get { return ObjectForKeyedSubscript (goal); }
+			get { return ObjectForKeyedSubscript (goal)!; }
```
Comment (tj-devel709) [inline src/GameplayKit/GKBehavior.cs:23]: From here: https://developer.apple.com/documentation/gameplaykit/gkbehavior/1388723-objectforkeyedsubscript?language=objc

`Return Value
The weight to be applied to the goal’s influence on an agent’s speed and direction, or 0.0 if the goal is not in the behavior.`

Comment (mandel-macaque) [reply src/GameplayKit/GKBehavior.cs:23]: Why do we need the !? I'm looking at the bindings and we have:
```
NSNumber ObjectForKeyedSubscript (GKBehavior behavior);
```

There is no nullable value, is there?

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

File: src/GameplayKit/GKBehavior.cs (hunk 1 above)
Comment (tj-devel709) [reply src/GameplayKit/GKBehavior.cs:23]: @mandel-macaque were you looking at this one inside GKCompositeBehavior https://github.com/xamarin/xamarin-macios/blob/main/src/gameplaykit.cs#L336?
This one refers to the ObjectForKeyedSubscript inside GKBehavior https://github.com/xamarin/xamarin-macios/blob/main/src/gameplaykit.cs#L227 which has a nullable return

**Review by rolfbjarne — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]
After Manuel's 👍

File: src/GameplayKit/GKBehavior.cs (hunk 1 above)
Comment (tj-devel709) [reply src/GameplayKit/GKBehavior.cs:23]: Note for future TJ: 
The tests show the return value is indeed null when the goal is not in the behavior, add that if this value returns null, throw an ArgumentOutOfRangeException, and file a radar with Apple that the APIs behavior is different than their implementation

Comment (tj-devel709) [reply src/GameplayKit/GKBehavior.cs:23]: Note for future TJ: 
The tests show the return value is indeed null when the goal is not in the behavior, add that if this value returns null, throw an ArgumentOutOfRangeException, and file a radar with Apple that the APIs behavior is different than their implementation

**Comment by mandel-macaque** [comment]
The errors are because the branch is behinds main and is using the main yaml. merging should fix the issue.

**Review by mandel-macaque — APPROVED** [review]

**Pull request merged**